portal-impl/src/com/liferay/portal/repository/external/LegacyExternalRepositoryDefiner.java

```diff
@@ -55,29 +53,9 @@ public RepositoryConfiguration getRepositoryConfiguration() {
 				return _repositoryConfiguration;
 			}
 
-			BaseRepository baseRepository =
```

Comment:
Hi @dantewang @ChiZelin ,
Do we have any replacement for this part of code? or why you think this logic is safe to remove?

Tina

Reply:
@tinatian There is a replacement, will double check.

Reply:
@tinatian The JavaDoc of the deprecated method says

```
	/**
	 * @deprecated As of Wilberforce (7.0.x), replaced by {@link
	 *             com.liferay.portal.kernel.repository.registry.RepositoryDefiner#getSupportedParameters(
	 *             )}
	 */
	@Deprecated
	public String[][] getSupportedParameters();
```

But the actual replacement should be `RepositoryDefiner.getRepositoryConfiguration().getParameters()`.

With my understanding, the new API wants devs to implement `RepositoryDefiner` whenever they implement a `Repository`. And the parameters should be directly provided into the `RepositoryConfiguration` in `RepositoryDefiner`, thus `getSupportedParameters()` was deprecated. For example, see [CMISWebServicesRepositoryDefiner
](https://github.com/liferay/liferay-portal/blob/743455aa37f43d0b6fbdf68581b7b4f4f84b03a5/modules/apps/document-library/document-library-repository-cmis-impl/src/main/java/com/liferay/document/library/repository/cmis/internal/CMISWebServicesRepositoryDefiner.java#L32)

I think the class modified here (`LegacyExternalRepositoryDefiner`) is a bridge between `Repository` still using the old API (implements `getSupportedParameters()` but doesn't have `RepositoryDefiner`) and the new API. If we remove `getSupportedParameters()`, we introduce a breaking change here and force devs to use the new API, then this class should be deprecated.

What do you think?

Reply:
@tinatian After a second thought I think removal of `getSupportedParameters()` should only be done when they want to completely drop support for what they call "legacy repository", I will ask Leon to exclude this change and resend.